src/bvals/cc/sparse_flux_correction_cc_in_one.cpp

```diff
@@ -0,0 +1,335 @@
+    PARTHENON_REQUIRE(buf.GetState() == BufferState::stale, "Not sure how I got here.");
+#ifdef MPI_PARALLEL
+    Kokkos::fence();
+#endif
+    buf.Send();
+    return LoopControl::cont;
+  });
+
+  Kokkos::Profiling::popRegion();
+  return TaskStatus::complete;
+}
+
+TaskStatus StartReceiveSparseFluxCorrectionBuffers(std::shared_ptr<MeshData<Real>> &md) {
+  Kokkos::Profiling::pushRegion("Task_ReceiveFluxCorrectionBuffers");
+  Mesh *pmesh = md->GetMeshPointer();
+  ForEachBoundary<BoundaryType::reflux_recv>(
+      md, [&](sp_mb_t pmb, sp_mbd_t rc, nb_t &nb, const sp_cv_t v) {
+        PARTHENON_DEBUG_REQUIRE(
+            pmesh->boundary_comm_reflux_map.count(ReceiveKey(pmb, nb, v)) > 0,
+            "Boundary communicator does not exist");
+        auto &buf = pmesh->boundary_comm_reflux_map[ReceiveKey(pmb, nb, v)];
+        buf.TryStartReceive();
+      });
+  Kokkos::Profiling::popRegion();
+  return TaskStatus::complete;
+}
+
+TaskStatus ReceiveSparseFluxCorrectionBuffers(std::shared_ptr<MeshData<Real>> &md) {
+  Kokkos::Profiling::pushRegion("Task_ReceiveFluxCorrectionBuffers");
+  bool all_received = true;
+  Mesh *pmesh = md->GetMeshPointer();
+  ForEachBoundary<BoundaryType::reflux_recv>(
+      md, [&](sp_mb_t pmb, sp_mbd_t rc, nb_t &nb, const sp_cv_t v) {
+        PARTHENON_DEBUG_REQUIRE(
+            pmesh->boundary_comm_reflux_map.count(ReceiveKey(pmb, nb, v)) > 0,
+            "Boundary communicator does not exist");
+        auto &buf = pmesh->boundary_comm_reflux_map[ReceiveKey(pmb, nb, v)];
+        all_received = all_received && buf.TryReceive();
+      });
+
+  Kokkos::Profiling::popRegion();
+
+  if (all_received) return TaskStatus::complete;
+  return TaskStatus::incomplete;
+}
+
+TaskStatus SetFluxCorrections(std::shared_ptr<MeshData<Real>> &md) {
+  Kokkos::Profiling::pushRegion("SetFluxCorrections");
+
+  Mesh *pmesh = md->GetMeshPointer();
+
+  ForEachBoundary<BoundaryType::reflux_recv>(
+      md, [&](sp_mb_t pmb, sp_mbd_t rc, nb_t &nb, const sp_cv_t v) {
+        PARTHENON_DEBUG_REQUIRE(
+            pmesh->boundary_comm_reflux_map.count(ReceiveKey(pmb, nb, v)) > 0,
+            "Boundary communicator does not exist");
+        auto &buf = pmesh->boundary_comm_reflux_map[ReceiveKey(pmb, nb, v)];
+
+        // Check if this boundary requires flux correction
+        if ((!v->IsAllocated()) || buf.GetState() == BufferState::received_null) {
+          buf.Stale();
+          return LoopControl::cont;
+        }
+
+        // Need to caculate these bounds based on mesh position
+        // Average fluxes over area and load buffer
+        IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::interior);
+        IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::interior);
+        IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::interior);
+
+        int ks = kb.s;
+        int js = jb.s;
+        int is = ib.s;
+        int ke = kb.e;
+        int je = jb.e;
+        int ie = ib.e;
```

Comment:
as above, these are fixed and identical for all blocks, aren't they?

Reply:
I would defer this to the flux correction *in one* update.